app/policies/comment_policy.rb

```diff
@@ -0,0 +1,17 @@
+# Pundit class to control authorization for comment actions.
+class CommentPolicy < ApplicationPolicy
+  # only signed up users can comment a book
+  def create?
+    user.present?
+  end
+
+  # only signed up users can update a comment
+  def update?
+    user.present?
```

Comment:
No! this will return a not found error and that's not the issue here, the problem is that the user is not authorized to do this change. So test for ownership inside the policy and query from all Comments